### PR TITLE
Add tests for book, categories

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: src/test/java/application/onlinebookstore/contoller/BookControllerTest.java [hunk 1]
```diff
@@ -1,5 +1,121 @@
...
+        assertNotNull(actual);
+        assertNotNull(actual.getId());
+        assertEquals(expected.getTitle(), actual.getTitle());
+        assertEquals(expected.getPrice(), actual.getPrice());
+    }
+
+    @Test
+    @Sql(scripts = "classpath:database/book/add-test-books-and-categories.sql")
+    @DisplayName("Get book by title")
+    void getBook_ValidRequestDto_Success() throws Exception {
```
Comment (vovasalyha) [inline src/test/java/application/onlinebookstore/contoller/BookControllerTest.java:103]: isn't there any failed cases possible?

File: src/test/java/application/onlinebookstore/contoller/CategoryControllerTest.java [hunk 2]
```diff
@@ -1,4 +1,108 @@
...
+                .readValue(result.getResponse().getContentAsString(), CategoryDto.class);
+        assertNotNull(actual);
+        assertNotNull(actual.getId());
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.getDescription(), actual.getDescription());
+    }
+
+    @Test
+    @DisplayName("Get category by name")
+    void getCategory_ValidRequestDto_Success() throws Exception {
```
Comment (vovasalyha) [inline src/test/java/application/onlinebookstore/contoller/CategoryControllerTest.java:90]: add edge cases

File: src/test/resources/database/delete-all-from-db.sql [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
+DO $$
+    DECLARE
+        table_name text;
+    BEGIN
+        FOR table_name IN (SELECT table_name FROM information_schema.tables WHERE table_schema = 'test' AND table_type = 'BASE TABLE')
+            LOOP
+                EXECUTE 'DROP TABLE IF EXISTS ' || table_name || ' CASCADE';
+            END LOOP;
+    END;
+$$;
```
Comment (vovasalyha) [inline src/test/resources/database/delete-all-from-db.sql:10]: empty line

File: src/test/java/application/onlinebookstore/contoller/BookControllerTest.java [hunk 4]
```diff
@@ -1,5 +1,121 @@
...
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Container
+    private static final PostgreSQLContainer<?> postgresContainer =
+            CustomPostgreSQLContainer.getInstance();
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext,
+            @Autowired DataSource dataSource
```
Comment (vovasalyha) [inline src/test/java/application/onlinebookstore/contoller/BookControllerTest.java:55]: where is it used

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ, but see a couple of comments
